src/main/java/picard/fingerprint/HaplotypeProbabilities.java

```diff
@@ -190,17 +188,19 @@ void assertSnpPartOfHaplotype(final Snp snp) {
 
     public double scaledEvidenceProbabilityUsingGenotypeFrequencies(final double[] genotypeFrequencies) {
         final double[] likelihoods = getLikelihoods();
-        assert (genotypeFrequencies.length == likelihoods.length);
+        assert (genotypeFrequencies.length == nGeno);
```

Comment:
I don't think `assert` is what you want here, since they have no effect unless the code is run with the `-ea` flag. E.g. if you had a unit test that tried to trigger this assert, it would fail.